libcloud/utils/iso8601.py

```diff
@@ -114,6 +114,8 @@ def parse_date(datestring, default_timezone=UTC):
     default timezone specified in default_timezone is used. This is UTC by
     default.
     """
+    if datestring is None:
```

Comment:
I think it would be better to handle this differently (returning "random" date is almost never a good idea) - either return None in such scenario or throw an exception if datestring is None and handle that exception in CloudSigma driver code.